mlir/include/Gradient/IR/GradientOps.td

```diff
@@ -191,5 +192,128 @@ def VJPOp : Gradient_Op<"vjp", [
     let hasVerifier = 1;
 }
 
+def ForwardOp : Gradient_Op<"forward",
```

Comment:
Do the `FowardOp` and `ReverseOp` not need assembly formats?

Reply:
It's the opposite, they have their own assembly format. It is declared below:

```
  let hasCustomAssemblyFormat = 1;
```

and defined like this in a different file.

```cpp
void ForwardOp::print(OpAsmPrinter &p)
{
    function_interface_impl::printFunctionOp(p, *this, /*isVariadic=*/false,
                                             getFunctionTypeAttrName(), getArgAttrsAttrName(),
                                             getResAttrsAttrName());
}
```

I'm just reusing the implementation of the function op interface.